src/OpenSSL/crypto.py

```diff
@@ -248,7 +248,12 @@ def check(self):
         if self._only_public:
             raise TypeError("public key only")
 
-        if _lib.EVP_PKEY_type(self._pkey.type) != _lib.EVP_PKEY_RSA:
+        if hasattr(self._pkey, "type"):
+            key_type = self._pkey.type
+        else:
+            key_type = _lib.Cryptography_EVP_PKEY_id(self._pkey)
+
+        if _lib.EVP_PKEY_type(key_type) != _lib.EVP_PKEY_RSA:
```

Comment:
Yup, wasn't sure if we should refactor as part of this though. I can do that though.